api/v1beta1/target_test.go

```diff
@@ -4,18 +4,45 @@ package v1beta1
 
 import (
 	"errors"
+	"fmt"
 	"path/filepath"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var sampleNames = []string{
 	"foo", "bar", "baz", "boo", "default", "kube-one", "kube-two", "kube-three",
 }
 
+// matches is only used to unit-test the behavior of `match`
+func (t *Target) matches(names []string) ([]string, error) {
+	// Using a map to ensure each entry in the result is unique.
+	set := make(map[string]struct{})
+
+	for _, name := range names {
+		matched, err := t.match(name)
+		if err != nil {
+			return nil, err
+		}
+
+		if matched {
+			set[name] = struct{}{}
+		}
+	}
+
+	matchingNames := make([]string, 0, len(set))
+	for ns := range set {
+		matchingNames = append(matchingNames, ns)
+	}
+
+	return matchingNames, nil
```

Comment:
This is a good idea. Not in this test function specifically, but more generally in the `GetMatches` and `GetMatchesDynamic` methods.

Reply:
Actually, rather than choosing a sorting, the implentations just use the existing order in the Lists. If those are inconsistent, then it's more of a kubernetes API issue than something I think we should address. It could be more confusing to sort it some special way, when the user is expecting it to match whatever convention k8s has.